apollo-api/src/main/java/com/apollostack/api/Query.java

```diff
@@ -1,17 +1,34 @@
 package com.apollostack.api;
 
-import java.util.List;
+import java.util.Collections;
 import java.util.Map;
 
 /** TODO */
-public interface Query {
+public interface Query<V extends Query.Variables> {
```

Comment:
@felipecsl 
1. I will argue, especially if query paginated and you will have to pass cursor, limit and as example of Shopify schema I would say variables will be used 80% of times
2. `variables()` is not generic it returns `V extends Query.Variables`, so in case if query has variable you will get statically typed variables, yes internally we don't care but from user perspective would be nice to have, if user decides to get variables for query